src/app/store/slices/storage/storage.thunks/downloadItemsThunk.ts

```diff
@@ -69,6 +79,126 @@ export const downloadItemsThunk = createAsyncThunk<void, DriveItemData[], { stat
   },
 );
 
+export const downloadItemsAsZipThunk = createAsyncThunk<void, DriveItemData[], { state: RootState }>(
+  'storage/downloadItemsAsZip',
+  async (items: DriveItemData[], { rejectWithValue }) => {
+    const errors: unknown[] = [];
+    const downloadProgress: number[] = [];
+    const abortController = new AbortController();
+    const formattedDate = date.format(new Date(), 'DD/MM/YYYY - HH:mm');
+    const folderName = `Internxt (${formattedDate})`;
+    const folder = new FlatFolderZip(folderName, {});
+
+    const user = localStorageService.getUser();
+    if (!user) throw new Error('User not found');
+
+    const taskId = tasksService.create<DownloadFileTask>({
+      action: TaskType.DownloadFile,
+
+      showNotification: true,
+      stop: async () => {
+        abortController.abort();
+        folder.abort();
+      },
+      cancellable: true,
+
+      file: {
+        name: folderName,
+        type: 'zip',
+      },
+    });
+
+    tasksService.updateTask({
+      taskId,
+      merge: {
+        status: TaskStatus.InProcess,
+      },
+    });
+
+    const calculateProgress = () => {
+      const totalProgress = downloadProgress.reduce((previous, current) => {
+        return previous + current;
+      });
+
+      return totalProgress / downloadProgress.length;
+    };
+
+    const updateProgressCallback = (progress: number) => {
+      tasksService.updateTask({
+        taskId,
+        merge: {
+          progress,
+        },
+      });
+    };
+
+    items.forEach((_, index) => {
+      downloadProgress[index] = 0;
+    });
+
+    for (const [index, driveItem] of items.entries()) {
+      try {
+        if (driveItem.isFolder) {
+          await folderService.downloadFolderAsZip(
+            driveItem.id,
+            driveItem.name,
+            (progress) => {
+              downloadProgress[index] = progress;
+              updateProgressCallback(calculateProgress());
+            },
+            { destination: folder, closeWhenFinished: false },
+          );
+          downloadProgress[index] = 1;
+        } else {
+          const fileStream = await downloadFile({
+            fileId: driveItem.fileId,
+            bucketId: driveItem.bucket,
+            creds: {
+              user: user.bridgeUser,
+              pass: user.userId,
+            },
+            mnemonic: user.mnemonic,
+            options: {
+              abortController,
+              notifyProgress: (totalBytes, downloadedBytes) => {
+                const progress = downloadedBytes / totalBytes;
+
+                downloadProgress[index] = progress;
+
+                updateProgressCallback(calculateProgress());
+              },
+            },
+          });
+
+          folder.addFile(`${driveItem.name}.${driveItem.type}`, fileStream);
```

Comment:
We have implemented a cache system. If we add it here the speed of the download will be increased. 
Here's a suggestion, I think this will work well.

```suggestion
          const lruFilesCacheManager = await LRUFilesCacheManager.getInstance();
          const cachedFile = await lruFilesCacheManager.get(driveItem.id.toString());
          const isCachedFileOlder = checkIfCachedSourceIsOlder({ cachedFile, file: driveItem });

          let fileStream;
          if (cachedFile?.source && !isCachedFileOlder) {
            const blob = new Blob([new Uint8Array([1, 2, 3])], { type: 'application/octet-stream' });
            fileStream = blob.stream();
            updateProgressCallback(calculateProgress());
          } else {
            fileStream = await downloadFile({
              fileId: driveItem.fileId,
              bucketId: driveItem.bucket,
              creds: {
                user: user.bridgeUser,
                pass: user.userId,
              },
              mnemonic: user.mnemonic,
              options: {
                abortController,
                notifyProgress: (totalBytes, downloadedBytes) => {
                  const progress = downloadedBytes / totalBytes;

                  downloadProgress[index] = progress;

                  updateProgressCallback(calculateProgress());
                },
              },
            });
          }

          folder.addFile(`${driveItem.name}.${driveItem.type}`, fileStream);
```

These two lines:
```
const lruFilesCacheManager = await LRUFilesCacheManager.getInstance();
const cachedFile = await lruFilesCacheManager.get(driveItem.id.toString());
```

could be changes for this one:
`  const cachedFile = await getDatabaseFileSourceData({ fileId: file.id });`

when [this PR](https://github.com/internxt/drive-web/pull/574) has been merged.